planet/src/main/webapp/js/placesItinerary.js

```diff
@@ -13,55 +13,104 @@
+        const placesContainer = document.getElementById('places');
+        const placeElement = createPlaceElement(place);
+        placesContainer.appendChild(placeElement);
+    }    
+}
+
+function createPlaceElement(place) {
     const placeElement = document.createElement('div');
     placeElement.setAttribute('class', 'card place yellow lighten-4');
-    placeElement.setAttribute('id', 'place-' + ref);
+    placeElement.setAttribute('id', 'place-' + place.place_id);
 
+    // Create the add button
     const addButton = document.createElement('div');
     addButton.setAttribute('class', 'place-button');
     
-    const eventFromThisPlace = document.getElementById(ref);
+    const eventFromThisPlace = document.getElementById(place.place_id);
     if (eventFromThisPlace) {
         addButton.innerHTML = `<i class="material-icons small">playlist_add_check</i>`;
     }else {
-        addButton.setAttribute('onclick', 'openAddPlaceForm(event, "' + ref + '")');
+        addButton.setAttribute('onclick', 'openAddPlaceForm(event, "' + place.place_id + '")');
         addButton.innerHTML = `<i class="material-icons small">playlist_add</i>`;
     }
     placeElement.appendChild(addButton);
 
-    placeElement.innerHTML += 
-        `<div class="card-content">
-          <span class="card-title">` + name + `</span>
-          <p>` + address +`</p>
-        </div>`;
+    // Create place name as the card title
+    const placeElementContent = document.createElement('div');
+    placeElementContent.setAttribute('class', 'card-content');
+    placeElementContent.innerHTML += `<span class="card-title">` + place.name + `</span>`;
+
+    // Add the rest of information if they exist
+    if (place.rating) {
+        placeElementContent.appendChild(createParagraphElement(place.rating));    
+    }
+    if (place.photos) {
+        const placePhotoSrc = place.photos[0].getUrl({maxWidth:400, maxHeight:200});
+        const placePhotoElement = document.createElement('img');
+        placePhotoElement.src = placePhotoSrc;
+        placeElementContent.appendChild(placePhotoElement);    
+    }
+    if (place.formatted_address) {
+        placeElementContent.appendChild(createParagraphElement(place.formatted_address));    
+    }
+    if (place.opening_hours) {
+        place.opening_hours.weekday_text.forEach((day) => {
+            placeElementContent.appendChild(createParagraphElement(day));
+        }); 
+    }
+    if (place.url) {
+        const placeUrlElement = document.createElement('a');
+        placeUrlElement.href = place.url;
+        placeUrlElement.target = '_blank';
+        placeUrlElement.innerText = 'More';
+        placeElementContent.appendChild(placeUrlElement);    
+    }
+
+    placeElement.appendChild(placeElementContent);
     return placeElement;
 }
 
+function createParagraphElement(text){
+    const p = document.createElement('p');
+    p.innerText = text;
+    return p;
+}
+
 function openAddPlaceForm(event, ref) {
```

Comment:
what type is ref? Consider renaming this variable to something more descriptive. (I think it's a place id?)

Reply:
Fixed!